Realm/Realm/Logging/Logger.cs

```diff
@@ -147,21 +183,24 @@ public void Log(LogLevel level, string message)
         /// </summary>
         /// <param name="level">The criticality level for the message.</param>
         /// <param name="message">The message to log.</param>
-        protected abstract void LogImpl(LogLevel level, string message);
+        /// <param name="category">The category for the message.</param>
+        protected abstract void LogImpl(LogLevel level, string message, LogCategory? category = null);
```

Comment:
This is still a breaking change - the addition of the argument is the breaking part, the nullability or the default value are inconsequential (you can verify that by trying to keep the old implementation in any of the built-in loggers). Therefore, we should not make the category nullable as it's not really helping and we always provide category anyway.

Reply:
As per our discussion the latest [updates](https://github.com/realm/realm-dotnet/pull/3634/commits/55f074d0624d37de1be54c18d3c7f31b97f8d57c) are now using the `Logger` as the derived class, avoiding a breaking change. (The PR description has also been updated to highlight the reasoning for this.)